2007-03-15  Jakub Jelinek  <jakub@redhat.com>

	[BZ #4130]
	* login/utmp_file.c (setutent_file): Use O_LARGEFILE for
	open_not_cancel_2.
	(updwtmp_file): Likewise.

--- libc/login/utmp_file.c	14 Dec 2005 11:33:40 -0000	1.17
+++ libc/login/utmp_file.c	15 Mar 2007 20:06:17 -0000	1.18
@@ -140,11 +140,11 @@ setutent_file (void)
 
       file_name = TRANSFORM_UTMP_FILE_NAME (__libc_utmp_file_name);
 
-      file_fd = open_not_cancel_2 (file_name, O_RDWR);
+      file_fd = open_not_cancel_2 (file_name, O_RDWR | O_LARGEFILE);
       if (file_fd == -1)
 	{
 	  /* Hhm, read-write access did not work.  Try read-only.  */
-	  file_fd = open_not_cancel_2 (file_name, O_RDONLY);
+	  file_fd = open_not_cancel_2 (file_name, O_RDONLY | O_LARGEFILE);
 	  if (file_fd == -1)
 	    return 0;
 	}
@@ -459,7 +459,7 @@ updwtmp_file (const char *file, const st
   int fd;
 
   /* Open WTMP file.  */
-  fd = open_not_cancel_2 (file, O_WRONLY);
+  fd = open_not_cancel_2 (file, O_WRONLY | O_LARGEFILE);
   if (fd < 0)
     return -1;
 
